ui/plugin-system/src/components/QueryInspector.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  Box,
+  Button,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+  Stack,
+} from '@mui/material';
+import { useQueryClient } from '@tanstack/react-query';
+import { TimeSeriesQueryDefinition, UnknownSpec } from '@perses-dev/core';
+import { useActiveDatasourceHeaders, useActiveTimeSeriesQueries, useTimeRange } from '../runtime';
+
+export interface WarningDisplay {
+  query: string;
+  header: string;
+  summary: string;
+}
+
+interface QueryInspectorProps {
+  showTotalQueries?: boolean;
+}
+
+export function QueryInspector(props: QueryInspectorProps) {
```

Comment:
This component is just for debugging, but the getQueryCache approach will be used in the custom banner I build internally